### PR TITLE
Add snippets to PubSub's javadoc, PubSubSnippets class and tests

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ public Sink updateSinkAsync(String sinkName, String datasetName)
   public Sink getSink(String sinkName) {
     // [START getSink]
     Sink sink = logging.getSink(sinkName);
-    if (sink != null) {
+    if (sink == null) {
```
Comment (lesv) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java]: Sigh - I need to check closer.  Sorry.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+import org.junit.rules.Timeout;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+public class ITPubSubSnippets {
```
Comment (lesv) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java]: I don't know if I'm more surprised you did this, or wondering if this is all that can be done?

**Comment by lesv** [comment]
LGTM

File: google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java (hunk 2 above)
Comment (mziccard) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java]: > I don't know if I'm more surprised you did this, or wondering if this is all that can be done?

Care to elaborate a bit more? :) Not sure what you mean

Comment (lesv) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java]: I'm happy & surprised to see integration tests on the snippets. -- Thank you.

Just wondering if there could be more of them?  Or if you covered as much as possible

**Comment by lesv** [comment]
LGTM

**Pull request merged**